internal/cmd/operator-sdk/pkgmantobundle/cmd.go

```diff
@@ -0,0 +1,360 @@
+$ operator-sdk pkgman-to-bundle packagemanifests --image-tag-base quay.io/example/etcd
+INFO[0000] Packagemanifests will be migrated to bundles in bundle directory
+INFO[0000] Creating bundle/bundle-0.0.1/bundle.Dockerfile
+INFO[0000] Creating bundle/bundle-0.0.1/metadata/annotations.yaml
+...
+
+# After running the above command, the bundles will be generated in 'bundle/' directory.
+$ tree bundle/
+bundle/
+├── bundle-0.0.1
+│   ├── bundle.Dockerfile
+│   ├── manifests
+│   │   ├── etcdcluster.crd.yaml
+│   │   ├── etcdoperator.clusterserviceversion.yaml
+│   ├── metadata
+│   │   └── annotations.yaml
+│   └── tests
+│       └── scorecard
+│           └── config.yaml
+└── bundle-0.0.2
+    ├── bundle.Dockerfile
+    ├── manifests
+    │   ├── etcdbackup.crd.yaml
+    │   ├── etcdcluster.crd.yaml
+    │   ├── etcdoperator.v0.0.2.clusterserviceversion.yaml
+    │   ├── etcdrestore.crd.yaml
+    ├── metadata
+    │   └── annotations.yaml
+    └── tests
+        └── scorecard
+            └── config.yaml
+
+Also, images for the both the bundles will be built with the following names: quay.io/example/etcd:0.0.1 and quay.io/example/etcd:0.0.2.
+`
+)
+
+type pkgManToBundleCmd struct {
+	// Input packagemanifest directory.
+	pkgmanifestDir string
+
+	// Optional flags for generating and building bundles.
+	outputDir string
+	baseImg   string
+	buildCmd  string
+}
+
+// NewCmd returns the pkgManToBundleCmd configured with the provided input options.
+func NewCmd() *cobra.Command {
+	p := pkgManToBundleCmd{}
+
+	pkgManToBundleCmd := &cobra.Command{
+		Use:     "pkgman-to-bundle <packagemanifestdir>",
+		Short:   "Migrates packagemanifests to bundles",
+		Long:    longHelp,
+		Example: examples,
+		PreRunE: func(cmd *cobra.Command, args []string) (err error) {
+			return p.validate(args)
+		},
+		RunE: func(cmd *cobra.Command, args []string) (err error) {
+			p.pkgmanifestDir = args[0]
+			return p.run()
+		},
+	}
+
+	pkgManToBundleCmd.Flags().StringVar(&p.outputDir, "output-dir", "bundles", "Directory to write bundle to.")
+	pkgManToBundleCmd.Flags().StringVar(&p.baseImg, "image-tag-base", "", "Base container image name for bundle image tags, "+
+		"ex. my.reg/foo/bar-operator-bundle will become my.reg/foo/bar-operator-bundle:${package-dir-name} for each child directory name in the packagemanifests directory")
+
+	// TODO(varsha): enable users to provide a template command so that it can be run in all child directories to build image.
+	pkgManToBundleCmd.Flags().StringVar(&p.buildCmd, "build-cmd", "", "Build command to be run for building images. By default 'docker build' is run.")
+
+	return pkgManToBundleCmd
+}
+
+// Generate the bundles from the provided packagemanifest directory.
+func (p *pkgManToBundleCmd) run() (err error) {
+
+	// error if output bundle directory already exists.
+	if _, err := os.Stat(p.outputDir); !os.IsNotExist(err) {
+		return fmt.Errorf("output directory: %s for bundles already exists", p.outputDir)
+	}
```

Comment:
```suggestion
	if _, err := os.Stat(p.outputDir); err != nil && errors.Is(err, os.ErrExist) {
		return fmt.Errorf("output directory: %s for bundles already exists", p.outputDir)
	} else if err != nil {
		return err
	}
```

Reply:
It shouldn't be erroring here if the path does not exist, and if the directory is already present there wouldn't be an error in `os.Stat`. I can make it to:
```go
if _, err = os.Stat(p.outputDir); !errors.Is(err, os.ErrNotExist) {
		return fmt.Errorf("output directory: %s for bundles already exists", p.outputDir)
}
```

Reply:
I'm pretty sure you explicitly have to check `errors.Is(err, os.ErrExist)`, since the set of all stat errors is not just exist + not exist errors.
```go
if _, err = os.Stat(p.outputDir); err != nil && errors.Is(err, os.ErrExist) {
		return fmt.Errorf("output directory: %s for bundles already exists", p.outputDir)
}
```
Then you can avoid handling the err if that condition fails by letting a downstream call emit a more specific error.